homeassistant/components/ecoal_boiler.py

```diff
@@ -0,0 +1,97 @@
+"""
+Component to control ecoal/esterownik.pl coal/wood boiler controller.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/ecoal_boiler/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['ecoaliface==0.4.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "ecoal_boiler"
+DATA_ECOAL_BOILER = 'data_' + DOMAIN
+
+DEFAULT_USERNAME = "admin"
+DEFAULT_PASSWORD = "admin"
+
+
+# Available pump ids with assigned HA names
+# Available as switches
+AVAILABLE_PUMPS = {
+    "central_heating_pump": "Central heating pump",
+    "central_heating_pump2": "Central heating pump2",
+    "domestic_hot_water_pump": "Central hot water pump",
+}
+
+# Available temp sensor ids with assigned HA names
+# Available as sensors
+AVAILABLE_SENSORS = {
+    "outdoor_temp": 'Outdoor temperature',
+    "indoor_temp": 'Indoor temperature',
+    "indoor2_temp": 'Indoor temperature 2',
+    "domestic_hot_water_temp": 'Domestic hot water temperature',
+    "target_domestic_hot_water_temp": 'Target hot water temperature',
+    "feedwater_in_temp": 'Feedwater input temperature',
+    "feedwater_out_temp": 'Feedwater output temperature',
+    "target_feedwater_temp": 'Target feedwater temperature',
+    "fuel_feeder_temp": 'Fuel feeder temperature',
+    "exhaust_temp": 'Exhaust temperature',
+}
+
+
+CONF_SWITCHES = 'switches'
+SWITCHES_SCHEMA = []
+for pump_id in AVAILABLE_PUMPS.keys():
+    SWITCHES_SCHEMA.append(vol.Optional(pump_id))
+
+CONF_SENSORS = 'sensors'
+SENSORS_SCHEMA = []
```

Comment:
See above.

Reply:
1. What "type" do you wish to have for temperature sensors? 
```
SENSORS = {
    TYPE_WHAT_HERE: ('Freeze Protect Temperature', 'mdi:thermometer', '°C'),
}
```
2. How should I convert
`SWITCH_SCHEMA = vol.Schema({vol.Optional(CONF_ZONE_RUN_TIME): cv.positive_int})`
to list switches offered by controller?

Or could you please provide correct schema, I will adapt current code to parse it?

Reply:
Type should be a lowercase snakecase name that matches what the sensor does.

Here's the sensor schema eg:
```py
SENSOR_SCHEMA = vol.Schema({
    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(AVAILABLE_SENSORS)):
        vol.All(cv.ensure_list, [vol.In(AVAILABLE_SENSORS)])
})
```
Here's the switch schema:
```py
SWITCH_SCHEMA = vol.Schema({
    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(AVAILABLE_PUMPS)):
        vol.All(cv.ensure_list, [vol.In(AVAILABLE_PUMPS)])
})
```
Adjust as necessary. Eg should we have a default at all or should the monitored conditions key be required?

Reply:
`AVAILABLE_SENSORS` dict looks good, so that doesn't need changing, unless you want to add more items than name.

Reply:
Please check recent version, as I am little confused with that style of config.  Config key `CONF_MONITORED_CONDITIONS` looks like not needed to me, but I assume it's common in other configs.
Default all for both pumps and sensors is OK.